dvc/repo/experiments/remove.py

```diff
@@ -70,6 +72,15 @@ def remove(  # noqa: C901, PLR0912
         exp_ref_list.extend(exp_refs(repo.scm, git_remote))
         removed = [ref.name for ref in exp_ref_list]
 
+    if keep:
+        all_exp_refs = exp_refs(repo.scm, git_remote)  # Get all experiments
```

Comment:
can something like:

`exp_ref_list  = all_exp_refs - exp_ref_list`

work in this case

I feel we don't need the `selected_exp_names` here

Reply:
correct. 
I had to add __eq__ and __hash__ methods to ExpRefInfo to make this nicer syntax work, and make sets out of what the generators yield